examples/refinement/refinement_demo.cc

```diff
@@ -63,9 +63,10 @@ int main(int argc, char **argv) {
   std::cout << "########################################" << std::endl;
   std::cout << "Initialization of data structure for mesh hierarchy"
             << std::endl;
-  std::shared_ptr<lf::mesh::hybrid2d::MeshFactory> mesh_factory_ptr =
-      std::make_shared<lf::mesh::hybrid2d::MeshFactory>(2);
-  lf::refinement::MeshHierarchy multi_mesh(mesh_ptr, mesh_factory_ptr);
+  std::unique_ptr<lf::mesh::hybrid2d::MeshFactory> mesh_factory_ptr =
+      std::make_unique<lf::mesh::hybrid2d::MeshFactory>(2);
+  lf::refinement::MeshHierarchy multi_mesh(mesh_ptr,
```

Comment:
User forced to rely on std::move(), but the compile won't allow anything else. Thus, it is safe.